frontend/controller/service-worker.js

```diff
@@ -62,28 +62,30 @@ sbp('sbp/selectors/register', {
 
     const pubsub = sbp('okTurtles.data/get', PUBSUB_INSTANCE)
     const existingSubscription = await registration.pushManager.getSubscription()
-    const messageToPushServerIfSocketOpen = (msgPayload) => {
+    const messageToPushServerIfSocketConnected = (msgPayload) => {
       // make sure the websocket client is not in the state of CLOSING, CLOSED before sending a message.
+      // if it is, attach a event listener to PUBSUB_RECONNECTION_SUCCEEDED sbp event.
       // (context: https://github.com/okTurtles/group-income/pull/1770#discussion_r1439005731)
 
       const readyState = pubsub.socket.readyState
-      const sendMsg = () => pubsub.socket.send(createMessage(
-        REQUEST_TYPE.PUSH_ACTION,
-        msgPayload
-      ))
+      const sendMsg = (socket) => {
+        (socket || pubsub.socket).send(createMessage(
```

Comment:
Better to just use `pubsub.socket` here I think. as the `socket` argument is going to refer to a custom event object instead of a socket

Reply:
@snowteamer that's right. updated.